corehq/apps/domain/forms.py

```diff
@@ -954,12 +956,9 @@ def save(self, domain):
 pwd_pattern = re.compile( r"([-\w]){"  + str(min_pwd) + ',' + str(max_pwd) + '}' )
 
 def clean_password(txt):
-    if len(txt) < min_pwd:
-        raise forms.ValidationError('Password is too short; must be at least %s characters' % min_pwd )
-    if len(txt) > max_pwd:
-        raise forms.ValidationError('Password is too long; must be less than %s characters' % max_pwd )
-    if not pwd_pattern.match(txt):
-        raise forms.ValidationError('Password may only contain letters, numbers, hyphens, and underscores')
+    strength = zxcvbn(txt, user_inputs=['commcare', 'hq', 'dimagi', 'commcarehq'])
+    if strength['score'] < 2:
+        raise forms.ValidationError(_('Password is not strong enough. Try making your password more complex.'))
```

Comment:
(Interesting point about password complexity over at [XKCD](https://xkcd.com/936/). See also [Diceware](https://en.wikipedia.org/wiki/Diceware), which is how I pick passwords to lock KeePass files these days.)